test/fakes/fakedatastore/fakedatastore.go

```diff
@@ -19,20 +18,22 @@ import (
 	"github.com/spiffe/spire/proto/spire/common"
 	spi "github.com/spiffe/spire/proto/spire/common/plugin"
 	"github.com/spiffe/spire/proto/spire/server/datastore"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
 const (
 	selectorKeySeparator = '|'
 )
 
 var (
-	ErrBundleAlreadyExists       = errors.New("bundle already exists")
-	ErrNoSuchBundle              = errors.New("no such bundle")
-	ErrAttestedNodeAlreadyExists = errors.New("attested node entry already exists")
-	ErrNoSuchAttestedNode        = errors.New("no such attested node entry")
-	ErrNoSuchRegistrationEntry   = errors.New("no such registration entry")
-	ErrNoSuchToken               = errors.New("no such token")
-	ErrTokenAlreadyExists        = errors.New("token already exists")
+	ErrBundleAlreadyExists       = status.Error(codes.NotFound, "bundle already exists")
```

Comment:
these "exists" errors seem more like codes.PreconditionFailed than codes.NotFound

Reply:
Maybe? Maybe I should do a closer pass as some of these since the real version returns zero values rather than errors in some NotFound cases.

I haven't read the code that's making use of the fake but it seems like there is use in it returning NotFound in the same cases as the real version.

Reply:
The fake should definitely be consistent with the SQL implementation for sure. If it's not, that needs to get fixed.

Reply:
Hmm, I was very liberal with this status code, not sure how it escaped me that "already exists" != "not found" :). Fixing.

Reply:
I've updated this, in some ways this is now inconsistent, and I'm not converting to FailedPrecondition in the real implementation. Would you like me to do that now?

Reply:
I'd be ok with either not having the fake return FailedPrecondition (yet) or updating the real implementation to return it in the right circumstances, as long as we remain consistent.